frontend/src/pages/ErrorsV2/ErrorInstance/ShowLogsButton.tsx

```diff
@@ -0,0 +1,79 @@
+import { Box, IconSolidLogs } from '@highlight-run/ui'
+import moment from 'moment'
+import { createSearchParams } from 'react-router-dom'
+
+import { useAuthContext } from '@/authentication/AuthContext'
+import { LinkButton } from '@/components/LinkButton'
+import { GetErrorInstanceQuery } from '@/graph/generated/operations'
+import { ReservedLogKey } from '@/graph/generated/schemas'
+import {
+	DEFAULT_LOGS_OPERATOR,
+	LogsSearchParam,
+	stringifyLogsQuery,
+} from '@/pages/LogsPage/SearchForm/utils'
+
+const getLogsLink = (data: GetErrorInstanceQuery | undefined): string => {
+	const errorObject = data?.error_instance?.error_object
+
+	if (!errorObject) {
+		return ''
+	}
+
+	const queryParams: LogsSearchParam[] = []
+	let offsetStart = 1
+	if (errorObject.session?.secure_id) {
+		queryParams.push({
+			key: ReservedLogKey.SecureSessionId,
+			operator: DEFAULT_LOGS_OPERATOR,
+			value: errorObject.session?.secure_id,
+			offsetStart: offsetStart++,
+		})
+	}
+	if (errorObject.trace_id) {
+		queryParams.push({
+			key: ReservedLogKey.TraceId,
+			operator: DEFAULT_LOGS_OPERATOR,
+			value: errorObject.trace_id,
+			offsetStart: offsetStart++,
+		})
+	}
+	const query = stringifyLogsQuery(queryParams)
+	const logCursor = errorObject.log_cursor
+	const params = createSearchParams({
+		query,
+		start_date: moment(errorObject.timestamp)
+			.add(-5, 'minutes')
+			.toISOString(),
+		end_date: moment(errorObject.timestamp).add(5, 'minutes').toISOString(),
+	})
+	if (logCursor) {
+		return `/${errorObject.project_id}/logs/${logCursor}?${params}`
+	} else {
+		return `/${errorObject.project_id}/logs?${params}`
+	}
+}
+
+type Props = {
+	data: GetErrorInstanceQuery | undefined
+}
+
+export const ShowLogsButton = ({ data }: Props) => {
+	const { isLoggedIn } = useAuthContext()
+
+	const logsLink = getLogsLink(data)
+
+	return (
+		<LinkButton
+			kind="primary"
+			emphasis="high"
+			disabled={!isLoggedIn || logsLink === ''}
```

Comment:
Not a blocker, but sort of makes me weary using the `===` comparison here. Is it possible to use `!logsLink` that would also catch null and undefined values?

Reply:
Good question. The return value of `getLogsLink` is a `string` so it can't be undefined or null.
https://github.com/highlight/highlight/blob/3363218b9b936c30d76fe083f6430b98caec7ea1/frontend/src/pages/ErrorsV2/ErrorInstance/ShowLogsButton.tsx#L15-L20

Reply:
For additional context, I fallback to an empty string intentionally because the `to` prop is required for `<LinkButton />`. It's somewhat of a non-graceful way of handling dynamic links when they're disabled. Here's another example of us doing the same thing:

https://github.com/highlight/highlight/blob/3363218b9b936c30d76fe083f6430b98caec7ea1/frontend/src/pages/Setup/IntegrationBar.tsx#L131-L137